packages/material-ui-unstyled/src/SwitchUnstyled/useSwitch.ts

```diff
@@ -0,0 +1,41 @@
+import { ChangeEvent, ChangeEventHandler } from 'react';
+import { unstable_useControlled as useControlled } from '@material-ui/utils';
+
+export interface UseSwitchProps {
+  checked?: boolean;
+  defaultChecked?: boolean;
+  disabled?: boolean;
+  onChange?: ChangeEventHandler<HTMLInputElement>;
+}
+
+export default function useSwitch(props: UseSwitchProps) {
+  const { checked: checkedProp, defaultChecked, disabled, onChange } = props;
+
+  const [checked, setCheckedState] = useControlled({
+    controlled: checkedProp,
+    default: Boolean(defaultChecked),
+    name: 'SwitchUnstyled',
+    state: 'checked',
+  });
+
+  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    // Workaround for https://github.com/facebook/react/issues/9023
+    if (event.nativeEvent.defaultPrevented) {
+      return;
+    }
+
+    setCheckedState(event.target.checked);
+    onChange?.(event);
```

Comment:
I'm trying to update these whenever I encounter them.

Reply:
Babel transpile them [a bit differently](https://babeljs.io/repl#?browsers=ie%2011%2C%20edge%20%3E%3D%2014%2C%20firefox%20%3E%3D%2052%2C%20chrome%20%3E%3D%2049%2C%20safari%20%3E%3D%2010%2C%20node%208.0%0A&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=PYOwwgFghiDmCmB-AdACngN3iALgSgG4g&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=env%2Creact&prettier=false&targets=&version=7.14.4&externalPlugins=). @eps1lon any preferences on this topic?

Reply:
> onChange?.(event);

:+1:  That's describing better what we want conceptually.

> if (onChange) onChange

:-1: This is not reasonable since it relies on JS implicit type conversions.